[WFLY-15542] Connector subsystem Jakarta migration

tadamski opened the pull request:
https://issues.redhat.com/browse/WFLY-15542

Depends on:
https://github.com/wildfly/wildfly/pull/14975

tadamski committed:
[WFLY-15542] Connector subsystem Jakarta migration

tadamski committed:
[WFLY-15825] Upgrade ironjacamar to 1.5.5.Final

yersan commented on ee-9/pom.xml:
```diff
@@ -416,6 +417,18 @@
                 </exclusions>
             </dependency>
 
+            <dependency>
+                <groupId>jakarta.security.auth.message</groupId>
+                <artifactId>jakarta.security.auth.message-api</artifactId>
+                <version>${version.jakarta.security.auth.message-api}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>*</groupId>
+                        <artifactId>*</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
```
This dependency should be `jakarta.authentication:jakarta.authentication-api:2.0.0` instead, which is already available in the pom, so I would not expect to introduce this variant (2.0.0-RC1) here.

tadamski replied:
fixed; I found out that this can be removed

darranl commented on ee-9/source-transform/connector/pom.xml:
```diff
@@ -0,0 +1,210 @@
+        Maintain separation between the artifact id and the version to help prevent
+        merge conflicts between commits changing the GA and those changing the V.
+        -->
+        <version>26.0.0.Final-SNAPSHOT</version>
```
I think this will need updating to the current version in use on main.

tadamski replied:
fixed

darranl commented:
@tadamski I think this one still needs some updates before we test again.

tadamski commented:
@darranl I plan to fix and rebase this PR after transactions migration is merged.

bstansberry commented:
@tadamski The transactions subsystem one is merged now.

tadamski commented:
@bstansberry I have rebased and fixed the PR; should work fine now

tadamski commented on ee-9/pom.xml:
```diff
@@ -771,6 +771,103 @@
+<!--                        <groupId>*</groupId>-->
+<!--                        <artifactId>*</artifactId>-->
+<!--                    </exclusion>-->
+<!--                </exclusions>-->
```
fixed

tadamski replied:
fixed

tadamski commented on ee-9/pom.xml:
```diff
@@ -771,6 +771,103 @@
+                        <artifactId>*</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
```
fixed

tadamski commented on ee-9/pom.xml:
```diff
@@ -1284,12 +1381,12 @@
+<!--                        <groupId>*</groupId>-->
+<!--                        <artifactId>*</artifactId>-->
+<!--                    </exclusion>-->
+<!--                </exclusions>-->
```
fixed

tadamski commented on ee-9/source-transform/connector/pom.xml:
```diff
@@ -0,0 +1,210 @@
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>wildfly-connector-jakarta</artifactId>
```
fixed

tadamski commented:
@yersan thanks for all the feedback - I have fixed the problems that you mentioned above

@fjuma @darranl IronJacamar still relies on PicketBox. Because of that, I had to temporary restore PicketBox dependency to make ee-9 connector compile:
https://github.com/tadamski/wildfly/commit/8cd99816b30a2db5d69a28831f9c1b739709c638#diff-31d9e3bb0671159c056658b3acb69396cbcf1a43a00db007c39c99817177c190R262

there is a integration layer that I think should be migrated to Elytron:
https://github.com/ironjacamar/ironjacamar/tree/main/core/impl/src/main/java/org/jboss/jca/core/security/picketbox

I have created https://issues.redhat.com/browse/JBJCA-1441 to deal with this issue.  

The question is whether it should be implemented before this PR is fixed. For now, I have updated the branch so that it can be merged before the fix.

yersan commented on ee-9/feature-pack/pom.xml:
```diff
@@ -779,6 +779,38 @@
+                    <groupId>org.jboss.ironjacamar</groupId>
+                    <artifactId>ironjacamar-common-api</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.jboss.ironjacamar</groupId>
+                    <artifactId>ironjacamar-common-impl-jakarta</artifactId>
+                </exclusion>
```
Hi @tadamski , you have to exclude the old non -jakarta versions from this feature pack.

tadamski replied:
hi @yersan, fixed

bstansberry commented on ee-9/pom.xml:
```diff
@@ -1425,6 +1521,18 @@
+                        <artifactId>*</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
```
This dependency no longer exists.

tadamski replied:
fixed

bstansberry commented:
Thanks @tadamski! :) And @yersan for the review
